sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -1,7 +1,50 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { RequestParameters } from "./common.js";
+import { PathParameterWithOptions, RequestParameters } from "./common.js";
+
+type QueryParameterStyle = "form" | "spaceDelimited" | "pipeDelimited";
+
+/**
+ * An object that can be passed as a query parameter, allowing for additional options to be set relating to how the parameter is encoded.
+ */
+interface QueryParameterWithOptions {
+  /**
+   * The value of the query parameter.
+   */
+  value: unknown;
+
+  /**
+   * If set to true, value must be an array. Setting this option to true will cause the array to be encoded as multiple query parameters.
+   * Setting it to false will cause the array values to be encoded as a single query parameter, with each value separated by a comma ','.
+   *
+   * For example, with `explode` set to true, a query parameter named "foo" with value ["a", "b", "c"] will be encoded as foo=a&foo=b&foo=c.
+   * If `explode` was set to false, the same example would instead be encouded as foo=a,b,c.
+   *
+   * Defaults to false.
+   */
+  explode?: boolean;
+
+  /**
+   * Style for encoding arrays. Three possible values:
+   * - "form": array values will be separated by a comma "," in the query parameter value.
+   * - "spaceDelimited": array values will be separated by a space (" ", url-encoded to "%20").
+   * - "pipeDelimited": array values will be separated by a pipe ("|").
+   *
+   * Defaults to "form".
+   */
+  style?: QueryParameterStyle;
+}
+
+function isPathParameterWithOptions(x: unknown): x is PathParameterWithOptions {
+  const value = (x as PathParameterWithOptions).value as any;
+  return (
+    value !== undefined && value.toString !== undefined && typeof value.toString === "function"
+  );
+}
+
+const isQueryParameterWithOptions: (x: unknown) => x is QueryParameterWithOptions =
```

Comment:
feels a little weird to overlap these checks since in `PathParameterWithOptions` we know the value must be a string (according to the types) whereas the query param need only be coerced into a string.